editor/src/core/shared/project-file-types.ts

```diff
@@ -543,12 +543,17 @@ export function isParsedJSONFailure(result: ParsedJSONResult): result is ParsedJ
 export type ParsedJSONResult = ParsedJSONSuccess | ParsedJSONFailure
 
 // Ensure this is kept up to date with clientmodel/lib/src/Utopia/ClientModel.hs.
-export type RevisionsStateType = 'PARSED_AHEAD' | 'CODE_AHEAD' | 'BOTH_MATCH'
+export type RevisionsStateType =
+  | 'PARSED_AHEAD'
+  | 'CODE_AHEAD'
+  | 'BOTH_MATCH'
+  | 'CODE_AHEAD_BUT_PLEASE_TELL_VSCODE_ABOUT_IT'
 
 export const RevisionsState = {
   ParsedAhead: 'PARSED_AHEAD',
   CodeAhead: 'CODE_AHEAD',
   BothMatch: 'BOTH_MATCH',
+  CodeAheadButPleaseTellVSCodeAboutIt: 'CODE_AHEAD_BUT_PLEASE_TELL_VSCODE_ABOUT_IT',
```

Comment:
Don't make the change I'm suggesting, but I can imagine this would be nicer if we did:

```typescript
{
  type: 'CODE_AHEAD',
  shouldTellVSCode: boolean
}
```

But what we have is better for making sure we enumerate the cases.

Reply:
That might actually be relevant to my comment (about `CODE_AHEAD` being able to replace `CODE_AHEAD_BUT_PLEASE_TELL_VSCODE_ABOUT_IT`). I think at this point that's more of a theoretical issue, but we could certainly look at this if it becomes a realised issue

Reply:
+1